src/components/switch/switch.stories.tsx

```diff
@@ -0,0 +1,176 @@
+    </>
+  );
+};
+
+export const Disabled = () => {
+  return (
+    <>
+      <Switch label="Disabled switch" disabled />
+      <Switch label="Disabled switch" disabled checked mt={2} />
+    </>
+  );
+};
+
+export const Required = Default.bind({});
+Required.args = {
+  label: "Terms and Conditions",
+  name: "required",
+  required: true,
+};
+
+export const Reversed = Default.bind({});
+Reversed.args = {
+  label: "Reversed switch",
+  name: "reversed",
+  reverse: false,
+};
+
+export const Loading = () => {
+  return (
+    <>
+      <Switch checked label="small on" size="small" loading />
+      <Switch label="small off" size="small" loading />
+      <Switch checked label="large on" size="large" loading />
+      <Switch label="large off" size="large" loading />
+    </>
+  );
+};
+
+export const WithLabelInline = () => {
+  return (
+    <>
+      <Switch label="With labelInline" labelInline />
+      <Switch
+        label="With labelInline and reversed"
+        labelInline
+        reverse={false}
+      />
+    </>
+  );
+};
+
+export const WithFieldHelp = () => {
+  return (
+    <>
+      <Switch
+        label="With fieldHelp"
+        fieldHelp="This text provides help for the input."
+      />
+      <Switch
+        label="With inline fieldHelp"
+        fieldHelp="This text provides help for the input."
+        fieldHelpInline
+      />
+    </>
+  );
+};
+
+export const WithLabelHelp = Default.bind({});
+WithLabelHelp.args = {
+  label: "With labelHelp",
+  labelHelp: "This text provides more information for the label.",
+  helpAriaLabel: "This text provides more information for the label.",
+};
+
+export const WithMargin = () => {
+  return (
+    <>
+      <Switch
+        label="With labelHelp"
+        labelHelp="This text provides more information for the label."
+        m={2}
+      />
+      <Switch
+        label="With labelHelp"
+        labelHelp="This text provides more information for the label."
+        m={4}
+      />
+      <Switch
+        label="With labelHelp"
+        labelHelp="This text provides more information for the label."
+        m="9px"
+      />
+    </>
+  );
+};
+
+export const Validation: StoryFn = (
+  args: Partial<SwitchProps> & { validation?: string | boolean }
+) => {
+  return (
+    <>
+      <Switch
+        error={args.validation}
+        label="Example switch (error)"
+        name="switch-error"
+        validationOnLabel={args.validationOnLabel}
+        tooltipPosition={args.tooltipPosition}
+      />
+      <Switch
+        warning={args.validation}
+        label="Example switch (warning)"
+        name="switch-warning"
+        validationOnLabel={args.validationOnLabel}
+        tooltipPosition={args.tooltipPosition}
+      />
+      <Switch
+        info={args.validation}
+        label="Example switch (info)"
+        name="switch-info"
+        validationOnLabel={args.validationOnLabel}
+        tooltipPosition={args.tooltipPosition}
+      />
+    </>
+  );
+};
```

Comment:
Sort of related to this previous comment, I think the use of `Template.bind({})` is fine here, since we don't have any internal state